src/Tribe/Main.php

```diff
@@ -1488,18 +1506,72 @@ public function issue_noindex() {
 				return;
 			}
 
-			// By default, we add a noindex tag for all month view requests and any other
-			// event views that are devoid of events
-			$add_noindex  = ( ! $wp_query->have_posts() || Month_View::get_view_slug() === $context->get( 'view' ) );
+			$view = $context->get( 'view' );
+
+			$start_date = ! empty( $wp_query->query[ 'eventDate' ] ) ? $wp_query->get( 'eventDate' ) : $context->get( 'now' );
+			$start_date = Tribe__Date_Utils::build_date_object( $start_date );
+
+			/**
+			 * Allow specific views to hook in and add their own calculated events.
+			 * This *bypasses* the cached query immediately after it.
+			 *
+			 * @since TBD
+			 *
+			 * @param ?Tribe__Repository|null $events     The events repository. False if not hooked in to.
+			 * @param DateTime                $start_date The start date (object) of the query.
+			 * @param Tribe__Context          $context    The current context.
+			 *
+			 */
+			$events = apply_filters( 'tec_events_noindex', null, $start_date, $context );
```

Comment:
Since this piece of code just cares about adding a no-index key or not, why not filter a `boolean` value instead of moving around the code a repository?

Reply:
A boolean "are there events?"

Reply:
By passing the repo we allow modifying it - ex: via FBAR rather than just redoing the query and overwriting the value